### PR TITLE
Edit recipe data layer

### PR DESCRIPTION
**Pull request by arnekuchenbecker**

### CODE REVIEW

**Comment by arnekuchenbecker** [comment]
Both ```RecipeRepository``` and ```ProjectRepository``` have grown quite large, which will become worse when adding in the network stuff. Created #119 to fix this by splitting them up into more managable classes

File: app/src/main/java/com/scouts/kitchenplaner/datalayer/daos/RecipeDAO.kt [hunk 1]
```diff
@@ -58,19 +64,76 @@ interface RecipeDAO {
         return recipeId
     }
 
+    @Query("UPDATE recipeEntity SET title = :newTitle WHERE id = :id")
+    suspend fun updateRecipeName(id: Long, newTitle: String)
+
+    @Query("UPDATE recipeEntity SET description = :newDesc WHERE id = :id")
+    suspend fun updateRecipeDescription(id: Long, newDesc: String)
+
```
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/datalayer/daos/RecipeDAO.kt:72]: Why is it not also a Query function but an update function (or vice versa?)

File: app/src/main/java/com/scouts/kitchenplaner/datalayer/repositories/RecipeRepository.kt [hunk 2]
```diff
@@ -131,14 +136,86 @@ class RecipeRepository @Inject constructor(
...
+        recipeDAO.increaseInstructionStepOrder(recipeID, index)
+        recipeDAO.insertInstructionStep(InstructionEntity(index, recipeID, instruction))
+    }
+
+    suspend fun deleteInstructionStep(recipeID: Long, index: Int) {
+        recipeDAO.deleteInstructionStep(InstructionStepIdentifierDTO(recipeID, index))
+        recipeDAO.decreaseInstructionStepOrder(recipeID, index)
+    }
+
+    suspend fun updateInstruction(recipeID: Long, index: Int, newInstruction: String) {
```
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/datalayer/repositories/RecipeRepository.kt:173]: updateInstructionStep

File: app/src/main/java/com/scouts/kitchenplaner/datalayer/daos/RecipeDAO.kt (hunk 1 above)
Comment (arnekuchenbecker) [reply app/src/main/java/com/scouts/kitchenplaner/datalayer/daos/RecipeDAO.kt:72]: ```@Query``` annotations can take any SQLite query as their parameter (i.e. you could also do Deletes and Inserts with them), the other annotations are essentially just Quality of Life stuff. For small partial updates (in this case just setting a string to a new value if the primary key matches) using an ```@Update``` annotation would mean having to define a DTO just for setting the description. While I do like having many classes this did seem a little excessive.

**Review by kittyahy — APPROVED** [review]

**Pull request merged**